### DISCUSSION TITLE
gguf-py PyPI release question

### DISCUSSION THREAD

**Discussion by PenutChen** [ggml-org/llama.cpp#8272] [General]
The latest release of gguf-py was on Dec 13, 2023. There are many changes compared to the current source. Most users usually use the source version of gguf-py. Recently, Hugging Face has integrated with the gguf format, but it depends on the PyPI version of gguf-py, which causes some compatibility issues (https://github.com/huggingface/transformers/issues/31725).

I think this makes it hard to decide which version to follow. New gguf models usually come out with the latest gguf-py, but third-party integrations can only use the old package on PyPI. I'm wondering if future versions of gguf-py will be released on PyPI?

**Comment by ggerganov** [comment]
I tried to run the PyPI workflow, but it no longer works:

https://github.com/ggerganov/llama.cpp/actions/runs/9820095645/job/27114309369#step:7:30

```
Uploading gguf-0.9.0-py3-none-any.whl
WARNING  Error during upload. Retry with the --verbose option for more details. 
ERROR    HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/        
         User 'monatis' does not have a verified primary email address. Please  
         add a verified primary email before attempting to upload to PyPI. See  
         https://pypi.org/help/#verified-email for more information.      
```

Pinging @monatis

**Comment by monatis** [comment]
Thanks @ggerganov will have a look at this in the morning.

**Reply by mgoin** [reply]
Hey @monatis thanks for looking into this, any update to share?

**Comment by monatis** [comment]
Verified email and enabled TFA. Will test the pipeline now.

**Comment by monatis** [comment]
Seems like my permissions purged --cannot make a push.

```
remote: Permission to ggerganov/llama.cpp.git denied to monatis.                                                        
fatal: unable to access 'https://github.com/ggerganov/llama.cpp.git/': The requested URL returned error: 403            
```

**Reply by ggerganov** [reply]
You are still a collaborator to the project - should be able to push

**Comment by monatis** [comment]
Ok needed to update credentials on this device. PR #8410
